### PR TITLE
Add rough support for new tx-types

### PR DESCRIPTION
**Pull request by StephenButtolph**

### CODE REVIEW

File: services/indexes/pvm/writer.go [hunk 1]
```diff
@@ -410,67 +401,84 @@ func (w *Writer) indexTransaction(ctx services.ConsumerCtx, blkID ids.ID, tx txs
...
+		baseTx = castTx.BaseTx.BaseTx
+		typ = models.TransactionTypeRemoveSubnetValidator
 	case *txs.TransformSubnetTx:
-		return nil
+		baseTx = castTx.BaseTx.BaseTx
+		typ = models.TransactionTypeTransformSubnet
 	case *txs.AddPermissionlessValidatorTx:
-		return nil
+		baseTx = castTx.BaseTx.BaseTx
+		outs = &avaxIndexer.AddOutsContainer{
```
Comment (patrick-ogrady) [inline services/indexes/pvm/writer.go:442]: I think this should only be defined if `castTx.Subnet == constants.PrimaryNetworkID`

File: services/indexes/pvm/writer.go [hunk 2]
```diff
@@ -410,67 +401,84 @@ func (w *Writer) indexTransaction(ctx services.ConsumerCtx, blkID ids.ID, tx txs
...
+			// TODO: What to do about the different rewards owners?
+			err = w.insertTransactionsRewardsOwners(ctx, txID, castTx.ValidatorRewardsOwner, baseTx, castTx.StakeOuts)
+			if err != nil {
+				return err
+			}
+		}
 	case *txs.AddPermissionlessDelegatorTx:
-		return nil
+		baseTx = castTx.BaseTx.BaseTx
+		outs = &avaxIndexer.AddOutsContainer{
```
Comment (patrick-ogrady) [inline services/indexes/pvm/writer.go:462]: I think this should only be defined if castTx.Subnet == constants.PrimaryNetworkID

File: services/indexes/pvm/writer.go (hunk 1 above)
Comment (patrick-ogrady) [reply services/indexes/pvm/writer.go:442]: I think there is some handling we should do here but for now let's treat just like a baseTx unless it is the same as a staking tx

Comment (StephenButtolph) [reply services/indexes/pvm/writer.go:442]: I don't see why this should be treated differently than an `ExportTx`

**Pull request merged**